agent/src/analysis/inventory_parser.py

```
@@ -0,0 +1,756 @@
+        InventoryEntry objects for each row
+
+    Raises:
+        ImportError: If pyarrow is not installed
+    """
+    try:
+        import pyarrow.parquet as pq
+    except ImportError:
+        raise ImportError(
+            "pyarrow is required for Parquet parsing. "
+            "Install with: pip install pyarrow"
+        )
```

Ruff (0.14.14)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)